src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs

```diff
@@ -269,54 +272,70 @@ private async Task<bool> EnsureCatalogAsync(CancellationToken cancellationToken)
 
         private async Task<IEnumerable<string>> GetLibraryVersionsAsync(string groupName, CancellationToken cancellationToken)
         {
-            IEnumerable<Asset> assets = await GetAssetsAsync(groupName, cancellationToken).ConfigureAwait(false);
+            JObject root = await GetLibraryGroupMetadataAsync(groupName, cancellationToken).ConfigureAwait(false);
+            if (root != null && root["versions"] is JToken versionsToken)
+            {
+                return versionsToken.Values<string>();
+            }
 
-            return assets.Select(a => a.Version);
+            return Array.Empty<string>();
         }
 
-        private async Task<IEnumerable<string>> GetLibraryIdsAsync(string groupName, CancellationToken cancellationToken)
+        private async Task<JObject> GetLibraryGroupMetadataAsync(string groupName, CancellationToken cancellationToken)
         {
-            IEnumerable<string> versions = await GetLibraryVersionsAsync(groupName, cancellationToken).ConfigureAwait(false);
+            string localFile = Path.Combine(_provider.CacheFolder, groupName, "metadata.json");
+            string url = string.Format(_metaPackageUrlFormat, groupName);
 
-            return versions.Select(v => LibraryIdToNameAndVersionConverter.Instance.GetLibraryId(groupName, v, _provider.Id));
+            try
+            {
+                string json = await _cacheService.GetMetadataAsync(url, localFile, cancellationToken).ConfigureAwait(false);
+                if (!string.IsNullOrEmpty(json))
+                {
+                    return JObject.Parse(json);
+                }
+            }
+            catch
+            {
+                throw new InvalidLibraryException(groupName, _provider.Id);
+            }
+
+            return null;
         }
 
-        private async Task<IEnumerable<Asset>> GetAssetsAsync(string groupName, CancellationToken cancellationToken)
+        private async Task<IEnumerable<string>> GetLibraryFilesAsync(string libraryName, string version, CancellationToken cancellationToken)
         {
-            var list = new List<Asset>();
-            string localFile = Path.Combine(_provider.CacheFolder, groupName, "metadata.json");
-            string url = string.Format(_metaPackageUrlFormat, groupName);
+            string localFile = Path.Combine(_provider.CacheFolder, libraryName, $"{version}-metadata.json");
```

Comment:
>son"); [](http://example.com/codeflow?start=102&length=6)

nit: extra newline

Reply:
Not seeing it, is this a CodeFlow artifact?

Reply:
ok, this isn't where I marked it, I still see it in the try block though

Reply:
Ah, got it